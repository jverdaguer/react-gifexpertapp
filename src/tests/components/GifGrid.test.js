import { shallow } from 'enzyme';
import GifGrid from '../../components/GifGrid';
import { useFetchGifs } from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs');

describe('Tests on Component GifGrid', () => {
    const category = 'Duke';
    test('should mount component <GifGrid />', () => {
        useFetchGifs.mockReturnValue({ data: [], loading: true });
        const wrapper = shallow(<GifGrid category={category} />);
        expect(wrapper).toMatchSnapshot();
    });

    test('should show images once loaded', () => {
        const imgs = [
            {
                id: 'ABC',
                title: 'Titulo Imagen',
                url: 'https://localhost/imagen.jpg',
            },
        ];
        useFetchGifs.mockReturnValue({ data: imgs, loading: false });
        const wrapper = shallow(<GifGrid category={category} />);
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('p').exists()).toBe(false); // should not have a <p> if loaded
        expect(wrapper.find('GifGridItem').length).toBe(imgs.length);
    });
});
