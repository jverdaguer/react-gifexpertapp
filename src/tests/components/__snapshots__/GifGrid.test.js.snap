// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests on Component GifGrid should mount component <GifGrid /> 1`] = `
<Fragment>
  <h3>
    Duke
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Cargando...
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;

exports[`Tests on Component GifGrid should show images once loaded 1`] = `
<Fragment>
  <h3>
    Duke
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="ABC"
      key="ABC"
      title="Titulo Imagen"
      url="https://localhost/imagen.jpg"
    />
  </div>
</Fragment>
`;
